starlette/middleware/cors.py

```diff
@@ -149,18 +149,18 @@ def preflight_response(self, request_headers: Headers) -> Response:
 
         message.setdefault("headers", [])
         headers = MutableHeaders(scope=message)
+        headers.update(self.simple_headers)
         origin = request_headers["Origin"]
         has_cookie = "cookie" in request_headers
 
         # If request includes any cookie headers, then we must respond
         # with the specific origin instead of '*'.
         if self.allow_all_origins and has_cookie:
-            self.simple_headers["Access-Control-Allow-Origin"] = origin
```

Comment:
Yup - this is flat-out-wrong as it stands. We shouldn't be mutating anything on the class.